src/lib/core/coio.cc

```diff
@@ -332,32 +341,13 @@ coio_readn_ahead_timeout(struct iostream *io, void *buf, size_t sz,
 	ssize_t nrd = coio_read_ahead_timeout(io, buf, sz, bufsiz, timeout);
 	if (nrd >= 0 && nrd < (ssize_t)sz) { /* EOF. */
 		errno = EPIPE;
-		tnt_raise(SocketError, sio_socketname(io->fd),
-			  "unexpected EOF when reading from socket");
+		diag_set(SocketError, sio_socketname(io->fd),
+			 "unexpected EOF when reading from socket");
+		return -1;
 	}
 	return nrd;
 }
 
-/*
- * FIXME: Rewrite coio_read_ahead_timeout() w/o C++ exceptions and
- * drop this function.
- */
-ssize_t
-coio_read_ahead_timeout_noxc(struct iostream *io, void *buf, size_t sz,
-			     size_t bufsiz, ev_tstamp timeout)
```

Comment:
It resolves #4881 (together with drop of `coio_write_timeout_noxc()` below).

Reply:
Linked the issue to the pull request. Thanks.